GeneratorInterface/GenFilters/plugins/PythiaFilterMotherSister.cc

```diff
@@ -66,17 +68,30 @@ bool PythiaFilterMotherSister::filter(edm::StreamID, edm::Event& iEvent, const e
                ++dau) {
             // find the daugther you're interested in
             if (abs((*dau)->pdg_id()) == abs(sisterID)) {
+              int failTrackPt = 0;
+              int failLeptonPt = 0;
+              // check pt of the nephews
+              for (HepMC::GenVertex::particle_iterator nephew = (*dau)->end_vertex()->particles_begin(HepMC::children);
+                   nephew != (*dau)->end_vertex()->particles_end(HepMC::children);
+                   ++nephew) {
+                int nephew_pdgId = abs((*nephew)->pdg_id());
+                if (minLeptonPt > 0. and (nephew_pdgId == 11 or nephew_pdgId == 13 or nephew_pdgId == 15))
+                  failLeptonPt += ((*nephew)->momentum().perp() < minLeptonPt);
+                if (minTrackPt > 0. and nephew_pdgId == 211)
+                  failTrackPt += ((*nephew)->momentum().perp() < minTrackPt);
+              }
```

Comment:
Hi Maria. Let me understand. Why is 11,13 and 15 generic. I know it is called lepton pT, but does this cover all needs. Even more about the pion. Can you clarify why this can be hard-coded?

Reply:
Hi, are you asking if the current logic covers all possible decay products of the sister particle? If so, not all possible cases are covered; this is ok for my specific case. In order to make this more general, I could implement the pdgid(s) of the nephew particle(s) as parameters, in addition to the corresponding pts. Please, let me know if this option is ok for you, and if you have other comments.

Reply:
Right. That was my proposal: I would add them as steerable parameters. Thanks!